civis/io/_tables.py

```diff
@@ -328,17 +328,17 @@ def dataframe_to_civis(df, database, table, api_key=None,
     ...                                      'scratch.df_table')
     >>> poller.result()
     """
-    with tempfile.NamedTemporaryFile() as f:
-        df.to_csv(f.name, index=False, **kwargs)
-        poll = csv_to_civis(f.name, database=database, table=table,
-                            api_key=api_key, max_errors=max_errors,
-                            distkey=distkey, sortkey1=sortkey1,
-                            sortkey2=sortkey2, delimiter=',',
-                            headers=headers, credential_id=credential_id,
-                            polling_interval=_DEFAULT_POLLING_INTERVAL,
-                            existing_table_rows=existing_table_rows,
-                            archive=archive)
-    return poll
+    buf = io.BytesIO()
+    # NOTE: TextIOWrapper is not thread-safe
```

Comment:
Summary of offline convo:

thread-safety shouldn't create any problems here and writing to the buffer could prevent an unnecessary copy.